core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -288,7 +297,7 @@ class ReplicaFetcherThread(name: String,
     partitionMap.foreach { case (topicPartition, partitionFetchState) =>
       // We will not include a replica in the fetch request if it should be throttled.
       if (partitionFetchState.isActive && !shouldFollowerThrottle(quota, topicPartition))
-        requestMap.put(topicPartition, new JFetchRequest.PartitionData(partitionFetchState.offset, fetchSize))
+        requestMap.put(topicPartition, new JFetchRequest.PartitionData(partitionFetchState.fetchOffset, partitionFetchState.logStartOffset, fetchSize))
```

Comment:
Ah I see your point now. You are right that we don't have to include it in `AbstractFetcherThread`. I have removed it from `PartitionFetchState` in `AbstractFetcherThread` and the code change in `ConsumerFetcherThread` and `AbstractFetcherThread` is much smaller now. Thanks!